homeassistant/components/brunt/cover.py

```diff
@@ -153,25 +169,46 @@ def is_closed(self):
         """Return true if cover is closed, else False."""
         return self.current_cover_position == CLOSED_POSITION
 
-    def update(self):
-        """Poll the current state of the device."""
-        try:
-            self._state = self._bapi.getState(thingUri=self._thing_uri).get("thing")
-            self._available = True
-        except (TypeError, KeyError, NameError, ValueError) as ex:
-            _LOGGER.error("%s", ex)
-            self._available = False
-
-    def open_cover(self, **kwargs):
+    async def async_open_cover(self, **kwargs):
         """Set the cover to the open position."""
-        self._bapi.changeRequestPosition(OPEN_POSITION, thingUri=self._thing_uri)
+        await self._async_update_cover(OPEN_POSITION)
 
-    def close_cover(self, **kwargs):
+    async def async_close_cover(self, **kwargs):
         """Set the cover to the closed position."""
-        self._bapi.changeRequestPosition(CLOSED_POSITION, thingUri=self._thing_uri)
+        await self._async_update_cover(CLOSED_POSITION)
 
-    def set_cover_position(self, **kwargs):
+    async def async_set_cover_position(self, **kwargs):
         """Set the cover to a specific position."""
-        self._bapi.changeRequestPosition(
-            kwargs[ATTR_POSITION], thingUri=self._thing_uri
+        await self._async_update_cover(kwargs[ATTR_POSITION])
+
+    async def _async_update_cover(self, position):
+        """Set the cover to the new position and wait for the update to be reflected."""
+        await self._bapi.async_change_request_position(
+            position, thingUri=self._thing.thingUri
         )
+        self.coordinator.update_interval = FAST_INTERVAL
+        await self.coordinator.async_request_refresh()
+
+    @callback
+    def _brunt_update_listener(self):
+        """Update the update interval after each refresh."""
+        if (
+            self.request_cover_position
+            == self._bapi.last_requested_positions[self._thing.thingUri]
+            and self.move_state == 0
+        ):
+            self.coordinator.update_interval = REGULAR_INTERVAL
+        else:
+            self.coordinator.update_interval = FAST_INTERVAL
+
+    @property
+    def device_info(self) -> dict:
+        """Return the device_info."""
+        return {
+            "identifiers": {(DOMAIN, self.unique_id)},
+            "name": self.name,
+            "via_device": None,
```

Comment:
I think `via_device` is optional, thus when you don't set it, I would just remove it :-).